### PR TITLE
Rekeying

### PR DESCRIPTION
**Pull request by ejohnstown**
1. Added rekeying during runtime, with a trigger function and highwater marks for number of bytes received and transmit.
2. Added a key generation function for producing a new RSA key for server identity.
3. Suggest updates from beta test.

### CODE REVIEW

**Comment by ejohnstown** [comment]
This is a release candidate for v1.0.0.

File: src/internal.c [hunk 1]
```diff
@@ -2303,11 +2501,47 @@ static int DoPacket(WOLFSSH* ssh)
...
+                        ret = WS_MEMORY_E;
+                    }
+                }
+
+                if (ssh->keyingState == KEYING_KEYED ||
+                    ssh->keyingState == KEYING_KEXINIT_SENT) {
+
+                    uint32_t idSz;
+
+                    wc_ShaUpdate(&ssh->handshake->hash,
```
Comment (toddouska) [inline src/internal.c]: Return code check?

File: src/internal.c [hunk 2]
```diff
@@ -2303,11 +2501,47 @@ static int DoPacket(WOLFSSH* ssh)
...
+                                 ssh->clientId, ssh->clientIdSz);
+
+                    idSz = (uint32_t)WSTRLEN(sshIdStr) - SSH_PROTO_EOL_SZ;
```
Comment (toddouska) [inline src/internal.c]: What is strlen() result is less than SSH_PROTO_EOL_SZ?

Comment (ejohnstown) [inline src/internal.c]: `sshIdStr` is a contant string.

File: src/internal.c (hunk 1 above)
Comment (ejohnstown) [inline src/internal.c]: I'll run through and fix the rest of those.

**Pull request merged**